app/controllers/concerns/csv_imports_controller_behavior.rb

```diff
@@ -0,0 +1,63 @@
+  module CsvImportsControllerBehavior
+    extend ActiveSupport::Concern
+    include Hydra::Controller::ControllerBehavior
+    include CurationConcerns::CurationConcernController
+
+    included do
+      self.curation_concern_type = form_class.model_class
+    end
+
+    def create
+      authenticate_user!
+      create_update_job
+      flash[:notice] = t('csv_create', application_name: view_context.application_name)
+      redirect_after_update
+    end
+
+    module ClassMethods
+      def form_class
+        ::CsvImportForm
+      end
+    end
+
+    protected
+
+      # Gives the class of the form.
+      # This overrides CurationConcerns
+      def form_class
+        self.class.form_class
+      end
+
+      def redirect_after_update
+        if uploading_on_behalf_of?
+          redirect_to sufia.dashboard_shares_path
+        else
+          redirect_to sufia.dashboard_works_path
+        end
+      end
+
+      def create_update_job
+        log = Sufia::BatchCreateOperation.create!(user: current_user,
+                                           operation_type: "CSV Import")
+
+        uploaded_files = params.fetch(:uploaded_files, [])
+        selected_files = params.fetch(:selected_files, {}).values
```

Comment:
If `uploaded_files` or `selected_files` hit the fallback case for `fetch` (returning [] or {}), then should we use that as a guard class and return? It looks like it's calling the `perform_later` method on the job regardless of there being files or not. Or, are files optional and this is desired behavior?